### ISSUE TITLE
Add in evaluation metric report publishing functionality

### ISSUE THREAD

**Issue by michaelkiper** [emit-sds/cover-class#42]
@marchett2 sent me [her code](https://github.jpl.nasa.gov/marchett/EMIT-FC/blob/main/pipeline/pipeline_classifier.py#L190#L210) for the current set of testing eval metrics. Since there will be so many hands on this code, a set of standardized reporting of performance for tracking and documenting will be very beneficial. As such, the `cover-class` repository should include an evaluation metric publishing function to generate a PDF of evaluation metrics and the various metadata for proper comparisons.

Training setup info:
1. Model architecture
2. Batch size
3. Optimizer
4. Optimization schedule
5. The subsampling info
6. The static data info (which files)
7. Simulation parameters
8. Version of the endmember library used
9. Timestamp

Training metrics:
1. A flexible set of training performance metrics - have the user pass in a dictionary of items to either JSON or turn into a chart (or allow them to pass in their own chart too)

Eval metrics (based on @marchett2 's code):
1. Confusion matrix
2. Per-class ROC plot (w/ AUC)
3. Missed class distribution
4. Per-class F1 scores (probably not varying betas)

TODO: ... survey if there are any more valuable perf metrics ... (e.g. Precision-Recall  AUC)
